grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/redisqueue/BlockingWorker.java

```diff
@@ -0,0 +1,208 @@
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+package ai.grakn.engine.tasks.manager.redisqueue;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.Timer.Context;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.concurrent.ExecutorService;
+import net.greghaines.jesque.Config;
+import net.greghaines.jesque.Job;
+import net.greghaines.jesque.json.ObjectMapperFactory;
+import net.greghaines.jesque.utils.PoolUtils;
+import net.greghaines.jesque.utils.PoolUtils.PoolWork;
+import static net.greghaines.jesque.utils.ResqueConstants.DATE_FORMAT;
+import static net.greghaines.jesque.utils.ResqueConstants.FAILED;
+import static net.greghaines.jesque.utils.ResqueConstants.INFLIGHT;
+import static net.greghaines.jesque.utils.ResqueConstants.PROCESSED;
+import static net.greghaines.jesque.utils.ResqueConstants.QUEUE;
+import static net.greghaines.jesque.utils.ResqueConstants.STARTED;
+import static net.greghaines.jesque.utils.ResqueConstants.STAT;
+import static net.greghaines.jesque.utils.ResqueConstants.WORKER;
+import static net.greghaines.jesque.utils.ResqueConstants.WORKERS;
+import static net.greghaines.jesque.worker.JobExecutor.State.NEW;
+import static net.greghaines.jesque.worker.JobExecutor.State.RUNNING;
+import static net.greghaines.jesque.worker.JobExecutor.State.SHUTDOWN;
+import static net.greghaines.jesque.worker.JobExecutor.State.SHUTDOWN_IMMEDIATE;
+import net.greghaines.jesque.worker.JobFactory;
+import static net.greghaines.jesque.worker.WorkerEvent.WORKER_ERROR;
+import static net.greghaines.jesque.worker.WorkerEvent.WORKER_POLL;
+import static net.greghaines.jesque.worker.WorkerEvent.WORKER_START;
+import static net.greghaines.jesque.worker.WorkerEvent.WORKER_STOP;
+import net.greghaines.jesque.worker.WorkerPoolImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import redis.clients.jedis.Jedis;
+import redis.clients.util.Pool;
+
+
+/**
+ * Queue implemented in redis
+ *
+ * @author Domenico Corapi
+ */
+public class BlockingWorker extends WorkerPoolImpl {
```

Comment:
How many workers are there per engine? I can't seem to see where this is configured.